src/mame/video/mrdo.cpp

```diff
@@ -185,13 +185,61 @@ void mrdo_state::mrdo_bgvideoram_w(offs_t offset, uint8_t data)
 	m_bg_tilemap->mark_tile_dirty(offset & 0x3ff);
 }
 
+uint8_t pal_u001;
```

Comment:
This needs to be encapsulated within the base class, not visible at global scope. In that case note the MAME convention of prefixing class variables, i.e. `m_pal_u001`.

Reply:
I moved it to class scope and renamed.

Reply:
it will also need registering to be saved in the machine_start, and initialized, probably in machine_reset

Reply:
Added m_pal_u001 save state handling.